server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreClusterStateRestoreIT.java

```diff
@@ -226,63 +116,170 @@ public void testFullClusterRestoreManifestFilePointsToInvalidIndexMetadataPathTh
         initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
 
         String prevClusterUUID = clusterService().state().metadata().clusterUUID();
+        String clusterName = clusterService().state().getClusterName().value();
 
         // Step - 2 Replace all nodes in the cluster with new nodes. This ensures new cluster state doesn't have previous index metadata
-        resetCluster(dataNodeCount, clusterManagerNodeCount);
-
-        String newClusterUUID = clusterService().state().metadata().clusterUUID();
-        assert !Objects.equals(newClusterUUID, prevClusterUUID) : "cluster restart not successful. cluster uuid is same";
-
+        internalCluster().stopAllNodes();
         // Step - 4 Delete index metadata file in remote
         try {
             Files.move(
                 segmentRepoPath.resolve(
-                    RemoteClusterStateService.encodeString(clusterService().state().getClusterName().value())
-                        + "/cluster-state/"
-                        + prevClusterUUID
-                        + "/index"
+                    RemoteClusterStateService.encodeString(clusterName) + "/cluster-state/" + prevClusterUUID + "/index"
                 ),
                 segmentRepoPath.resolve("cluster-state/")
             );
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
+        assertThrows(IllegalStateException.class, () -> addNewNodes(dataNodeCount, clusterManagerNodeCount));
+    }
+
+    public void testRemoteStateRestoreLongRunning() throws Exception {
+        int shardCount = randomIntBetween(1, 2);
+        int replicaCount = 1;
+        int dataNodeCount = shardCount * (replicaCount + 1);
+        int clusterManagerNodeCount = 3;
+
+        Map<String, Long> indexStats = initialTestSetup(shardCount, replicaCount, dataNodeCount, clusterManagerNodeCount);
 
-        // Step - 5 Trigger full cluster restore and validate fails
-        PlainActionFuture<RestoreRemoteStoreResponse> future = PlainActionFuture.newFuture();
-        restoreAndValidateFails(prevClusterUUID, future, IllegalStateException.class, "asdsa");
+        for (int i = 0; i < 100; i++) {
```

Comment:
we dont need to run so many iteration and also increasing test duration significantly. 1 iteration is sufficent. If there are any issues, it would turn up as flaky test.